component/component.go

```diff
@@ -291,3 +293,7 @@ type Generation interface {
 	// as the operation function itself such as Deploy or Release.
 	GenerationFunc() interface{}
 }
+
+type DeclaredResourcesResp struct {
```

Comment:
This struct needs comments!

Reply:
The `Resp` name in this is a bit weird to me based on the other components here. Maybe it should be `DelcaredResourcesResult` or `DeclaredResultsInfo` to match up with other component names?

Reply:
It's kind of a weird component, different from our others. This is to indicate to the plugin that it's not going to get anything out of this argument - it's an empty response vehicle that it's supposed to fill in. Kind of like how an http handler function takes a response as an argument.

Reply:
Added comments!